routing/pathfind_test.go

```diff
@@ -613,6 +613,88 @@ func TestKShortestPathFinding(t *testing.T) {
 	assertExpectedPath(t, paths[1], "roasbeef", "satoshi", "luoji")
 }
 
+func TestTotalAmountCalculation(t *testing.T) {
+
+	var sourceKey [33]byte
+	sourceVertex := Vertex(sourceKey)
+
+	const (
+		startingHeight = 100
+		finalHopCLTV   = 1
+	)
+
+	testCases := []struct { 
+		name 		    string
+		feeRates 	    []lnwire.MilliSatoshi
+		paymentAmount 	    lnwire.MilliSatoshi
+		expectedTotalAmount lnwire.MilliSatoshi
+	} {
+	{
+		// For a single hop payment, no fees are expected to be paid
+		name: "single hop", 
+		paymentAmount: 100000,
+		feeRates: []lnwire.MilliSatoshi{1000},
+		expectedTotalAmount: 100000,
+	}, {
+		// For a two hop payment, only the fee for the first hop
+		// needs to be paid
+		name: "two hop", 
+		paymentAmount: 100000,
+		feeRates: []lnwire.MilliSatoshi{1000, 1000},
+		expectedTotalAmount: 100100,
+	}, {
+		// A three hop payment where the first and second hop
+		// will both charge 1 msat. The fee for the first hop
+		// is actually slightly higher than 1, because the amount 
+		// t0 fwd also includes the fee for the second hop. This
+		// gets rounded down to 1.
+		name: "three hop", 
+		paymentAmount: 100000,
+		feeRates: []lnwire.MilliSatoshi{10, 10, 10},
+		expectedTotalAmount: 100002,
+	}, {
+		// A three hop payment where the fee of the first hop
+		// is slightly higher (11) than the fee at the second hop,
+		// because of the increase amount to forward.
+		name: "three hop with fee carry over", 
+		paymentAmount: 100000,
+		feeRates: []lnwire.MilliSatoshi{10000, 10000, 10000},
+		expectedTotalAmount: 102010,
```

Comment:
@halseth This test on master fails. newRoute on master returns 102000 for total amount.

Reply:
Good catch! Does it make sense to add this test + fix to a separate PR?

Reply:
Good idea. The fee limit changes are dependent on the previous (backward search) work, but this one is independent.